vllm/attention/backends/flashinfer.py

```diff
@@ -217,10 +247,14 @@ def forward(
                 self.kv_cache_dtype,
             )
 
+        query = query.contiguous(
+        )  # Flashinfer requires query to be contiguous
         if prefill_meta := attn_metadata.prefill_metadata:
-            # Prompt run.
-            assert prefill_meta.block_tables is not None
-            if kv_cache is None or prefill_meta.block_tables.numel() == 0:
+            # We will use flash attention for prefill
```

Comment:
why is that?  Can you comment? (also is it fundamental?)

Reply:
This only happens during the profiling phase, where the cache is initialized (not paged). We use flash attention for the profile run.